rstudio/rstudio
# Support new static content types publishing to Posit.cloud

### Intent

Addresses https://github.com/rstudio/rstudio-pro/issues/4541

The below assume that Posit cloud publishing is enabled in Global Options (true by default)

1. When the user attempts to publish any of the following content types, they should see the dialog to choose to publish to either Posit Connect or Posit Cloud [NEW functionality] followed by the dialog to publish with source code or finished content only [EXISTING functionality]
- RMarkdown site
- Quarto site
- Quarto book
- Quarto blog

_(Note you'll get an error when you actually try to publish as Posit Cloud has not yet been updated to support these content types just yet)_

2. When the user attempts to publish any of the following content types, they should see the dialog to choose to publish to either RPubs, Posit Connect, or Posit Cloud [EXISTING functionality] followed by the dialog to publish with source code or finished content only [NEW functionality]. _If Posit Cloud, the dialog should not make mention of "scheduling reports" as this is not supported by Cloud, only by Connect_
- RMarkdown single page
- Quarto single page
- Quarto presentation

3. When the user attempts to publish a plot, they should see the dialog to choose to publish to either RPubs, Posit Connect, or Posit Cloud [NEW functionality]. There's no dialog to publish with source code vs finished content in this case

### QA Notes

See above.
For Shiny apps, and multiple (static, non-website) Rmd pages, the user should not get asked about where to publish, but they should still see all relevant connected servers (shinyapps.io, posit.cloud, and posit connect for shiny apps; posit.cloud and posit connect for multiple Rmd pages)

### Checklist
